src/tape/ensemble.py

```diff
@@ -1587,3 +1587,326 @@ def sf2(self, sf_method="basic", argument_container=None, use_map=True):
             result = self.batch(calc_sf2, use_map=use_map, argument_container=argument_container)
 
             return result
+
+
+"""
+    The following package-level methods can be used to create a new Ensemble object 
+    by reading in the given data source.
+"""
+
+
+def read_pandas_dataframe(
+    source_frame,
+    object_frame=None,
+    dask_client=True,
+    column_mapper=None,
+    sync_tables=True,
+    npartitions=None,
+    partition_size=None,
+    **kwargs,
+):
+    """Read in Pandas dataframe(s) and return an ensemble object
+
+    Parameters
+    ----------
+    source_frame: 'pandas.Dataframe'
+        A Dask dataframe that contains source information to be read into the ensemble
+    object_frame: 'pandas.Dataframe', optional
+        If not specified, the object frame is generated from the source frame
+    dask_client: `dask.distributed.client` or `bool`, optional
+        Accepts an existing `dask.distributed.Client`, or creates one if
+        `client=True`, passing any additional kwargs to a
+            dask.distributed.Client constructor call. If `client=False`, the
```

Comment:
Extra indentation. Carries through to a weird rendering in the API doc:

https://tape--256.org.readthedocs.build/en/256/autoapi/tape/ensemble/index.html#tape.ensemble.read_pandas_dataframe

Same thing in the subsequent methods.

Reply:
Fixed (and surprised `black` didn't reformat these)